tracking/tracking_tests/reporter_tests.cpp

```diff
@@ -0,0 +1,62 @@
+#include "base/thread.hpp"
+#include "coding/traffic.hpp"
+
+#include "platform/location.hpp"
+#include "platform/socket.hpp"
+
+#include "std/cmath.hpp"
+
+#include "testing/testing.hpp"
+
+#include "tracking/reporter.hpp"
+
+namespace
+{
+template <class Condition>
+bool WaitCondition(Condition condition, size_t durationMs = 1000)
```

Comment:
Also, note, that if you'll destroy the reporter before checking what it wrote to the socket, note, that as it joins (not now, but should) the thread in the destructor, you do not need mutexes in the TestSocket at all, as there'll be a synchronization edge between reporter destructor and the following code! And you'll no need ReadServer() method, just implement an accessor to the m_output.